department-of-veterans-affairs/caseflow
# Jonathan/caseflow 4870 emo send to bva intake

Resolves https://vajira.max.gov/browse/CASEFLOW-4870

### Description
As a developer, to comply with the Education Pre-Docket requirements, I need understand the expected behavior of the EMO organization to develop the task actions that will be available for it.  The EMO organization will have the following task actions available:

- Send appeal to Regional Office
- Return to BVA Intake (because appeal is not an Education issue)
- Ready for Review (appeal is sent to BVA Intake to be docketed) 

We need to implement a new task action and subsequent modal for EMO users to successfully assign predocket appeals to BVA Intake as "Ready to Review" when the appeal is ready to be docketed.

### Acceptance Criteria
- [ ] "Send to Board Intake" task action is created
- [ ] "Ready to Review" modal is created
- [ ] Modal allows for the user to enter a description on where the documents are located and select between Mail Intake, VBMS and Other
- [ ] EMO can successfully send a predocket appeal to BVA Intake
- [ ] Upon success, show the appropriate success banner
- [ ] Upon fail, show a failure (most likely a Slim Error Alert in the modal)
- [ ] After assignment the EMO task has a status of "completed"
- [ ] The BVA Intake PreDocketTask should be updated from "on_hold" to "assigned"and the appeal shows in the BVA Intake "Ready for Review" tab

### Testing Plan
1. Run pre_docket_spec.rb
2. Go to /intake as a BVA Intake user
3. Submit an appeal with an Education issue and enter it into pre-docket.
4. Switch to an EMO user.
5. Go to the case details page for that appeal and copy the uuid.
6. Using the uuid, go into the rails console and view the Task tree. Verify that the EducationDocumentSearchTask status is set to 'assigned' and the parent PreDocketTask is set to 'on hold'.
7. On the case details page, click on the Actions dropdown menu and select 'Ready for Review'
8. Select a radio option and click the 'Submit' button.
9. Refresh the task tree and verify the Task statuses have changed. EducationDocumentSearchTask should be 'completed' and PreDocketTask should be 'assigned'


### User Facing Changes

 BEFORE|AFTER
 ---|---
x|![image](https://user-images.githubusercontent.com/98970951/164035770-e5450401-90be-467e-bfe8-14c21c6a6c60.png)
